### DISCUSSION TITLE
powertools-logging LoggingUtils supports removeKey and removeKeys

### DISCUSSION THREAD

**Discussion by seng-thebouqs** [aws-powertools/powertools-lambda-java#393] [Q&A] [answered]
Hi all, 
I have a question (or maybe a suggestion) about the power-tools-logging and pardon me if this is not the right place for this question. 
I recently come across this Lambda Power logging and its feature of changing the logging level via environment variable is a useful one for us. 
When I try to use it, I have coded to clear the custom key-pair values with ThreadContext.remove() and ThreadContext.clearAll() to avoid incorrect log output. I saw the power-tools logging LogginUtils contains method to appendKey() and appendKeys(), which in term calls ThreadContext.put() and putAll(). However, there is no wrapper methods to remove the keys. 
Hence, I have to still directly import the ThreadContext class to perform the remove operation in the code. 
It would be nice if LoggingUtils also include remove keys method so that my lambda code can completely rely on power-tools-logging for custom keys logging.

**Comment by pankajagrawal16** [comment, marked as answer]
Hi @seng-thebouqs Thanks for opening the issue. 

It would definitely makes sense to support a wrapper api to remove keys as well. Since this is one of the core utilities, and as part of our [tenants](https://awslabs.github.io/aws-lambda-powertools-java/#tenets) , we strive to remain consistent across runtimes. 

I just checked [python](https://awslabs.github.io/aws-lambda-powertools-python/latest/core/logger/#removing-additional-keys) version as well and it is supported. So yes, definitely, we will add it.
